cmd/kubectl-rbac-flatten/README.md

```diff
@@ -9,6 +9,67 @@ The kubectl-rbac-flatten command takes all the standard arguments for
 a Kubernetes command-line tool, and so can be used as a [kubectl
 plugin](https://kubernetes.io/docs/tasks/extend-kubectl/kubectl-plugins/).
 
+## Data Model
+
+The semantics of the collection of RBAC objects in a cluster is a
+collection of _privilege grants_, each of which has the following
+attributes.  Here we expand the semantics of the RBAC objects to
+atomic tuples; that is, each one identifies just one subject, verb,
+object, role-ism, and binding --- with the exception of wildcards.
+
+- **Subject**: who the privilege is granted to. A user (ordinary or
+  ServiceAccount) or group. Kubernetes authentication of a request
+  identifies the user and the set of groups that the user belongs
+  to. This set always includes either `system:authenticated` or
+  `system:unauthenticated`.
+
+- **Verb**: the action being allowed. When the _object_ is a
+  _resource_, the verb is one of: `create`, `update`, `patch`,
+  `delete`, `deletecollection`, `get`, `list`, `watch`.  When the
+  object is not a resource, the verb is one of: `get`, `put`,
+  `delete`, `post`.  The wildcard `*` grants usage of all verbs.
+
+- **Object**: the thing that the subject is allowed to invoke the verb on.
+  There are two cases here: API object, and "non-resource" URL path.
+
+    A non-resource URL path is just the path part of a URL; the schema
+    and authority of the authorized URL are implicitly those of the
+    apiserver.
+
+    An API object is identified by a "resource", "object name"
+    (misnamed "ResourceName" in the RBAC API), and possibly
+    "namespace".  A resource exists within an _api group_. A resource
+    is identified by a string of the form
+    `<base>.<apigroup>/<subresource>`, where `<base>` is the lowercase
+    plural way of identifying a kind of object. The `.<apigroup>` is
+    empty for the core API group.  The `/<subresource>` is optional.
+    Example resources: `nodes`, `jobs.batch`,
+    `clusterroles.rbac.authorization.k8s.io`,
+    `certificatesigningrequests/approval.certificates.k8s.io`. The
+    wildcard `*` can stand for all resources, for all namespaces, and
+    for all object names.
+
+    Each resource has a _scope_, which is either "cluster" or
+    "namespace".
+
+    Only a ClusterRoleBinding grants access to a
+    cluster-scoped resource. Nothing stops a RoleBinding from
+    referring to a ClusterRole that refers to cluster-scoped
+    resources, but that RoleBinding grants no actual privileges
+    regarding those resources.
+
+- **Role-ism**: the Role or ClusterRole object that defines the verb
+  and object.
+
+- **Binding**: the RoleBinding or ClusterRoleBinding ojbect that
+  associates the role-ism with the subject.
+
+## Query semantics
+
+This command can be directed to filter the reported grants to include
+only those that involve a given set of subjects, set of verbs, and/or
```

Comment:
The restriction by subject and verb are not implemented yet. The only implemented restriction so far is by "resource", which is the most interesting (as far as I know).